xarray/tests/test_backends.py

```diff
@@ -2652,6 +2652,36 @@ def test_open_mfdataset_does_same_as_concat(self, combine, opt, join):
                 ds_expect = xr.concat([ds1, ds2], data_vars=opt, dim="t", join=join)
                 assert_identical(ds, ds_expect)
 
+    def test_open_mfdataset_dataset_attr_by_coords(self):
+        """
+        Case when an attribute differs across the multiple files
+        """
+        with self.setup_files_and_datasets() as (files, [ds1, ds2]):
+            # Give the files an inconsistent attribute
+            for i, f in enumerate(files):
+                ds = open_dataset(f).load()
+                ds.attrs["test_dataset_attr"] = 10 + i
+                ds.close()
+                ds.to_netcdf(f)
+
+            with xr.open_mfdataset(files, combine="by_coords", concat_dim="t") as ds:
+                assert ds.test_dataset_attr == 10
+
+    def test_open_mfdataset_dataarray_attr_by_coords(self):
+        """
+        Case when an attribute of a member DataArray differs across the multiple files
+        """
+        with self.setup_files_and_datasets() as (files, [ds1, ds2]):
+            # Give the files an inconsistent attribute
+            for i, f in enumerate(files):
+                ds = open_dataset(f).load()
+                ds["v1"].attrs["test_dataarray_attr"] = i
+                ds.close()
+                ds.to_netcdf(f)
+
+            with xr.open_mfdataset(files, combine="by_coords", concat_dim="t") as ds:
+                assert ds["v1"].test_dataarray_attr == 0
```

Comment:
@TomNicholas do you think we should make this test pass (i.e. support combining `attrs` on DataArrays contained in the Dataset being loaded)?

Proposal for a fix:
* Support `open_mfdataset`-style attribute combining by upgrading the `combine_attrs` argument to `combine_nested`/`combine_by_coords` (and probably the other functions that have a `combine_attrs` argument) so it can be passed one of the `Dataset`s or `DataArray`s being combined/merged, and will then take the `attrs` from that.
* In `open_mfdataset` use the `attrs_file` argument to get the `Dataset` to take the `attrs` from, and pass that to the `combine_attrs` argument of `combine_nested`/`combine_by_coords`.